src/core/server/plugins/types.ts

```diff
@@ -209,6 +209,7 @@ export interface PluginInitializerContext<ConfigSchema = unknown> {
   };
   logger: LoggerFactory;
   config: {
+    globalConfig__deprecated$: Observable<Config>;
```

Comment:
Even if it's a deprecated accessor, I think we want to have a specialised type instead of the main root `Config` here, at least to avoid explicitly exposing the `Config#set` method. I would only keep `Config`'s `has` and `get` methods for what we expose here.

Reply:
I agree, maybe it's safer to expose read-only methods. Changing them now